src/query/xasl_to_stream.c

```diff
@@ -3234,6 +3246,22 @@ xts_process_xasl_node (char *ptr, const XASL_NODE * xasl)
     }
   ptr = or_pack_int (ptr, offset);
 
+  if (xasl->sq_cache)
+    {
+      ptr = or_pack_int (ptr, xasl->sq_cache->n_elements);
+    }
+  else
+    {
+      ptr = or_pack_int (ptr, 0);
+    }
+
+  offset = xts_save_sq_cache (xasl->sq_cache);
+  if (offset == ER_FAILED)
+    {
+      return NULL;
+    }
+  ptr = or_pack_int (ptr, offset);
+
```

Comment:
Please collect related codes in `xts_save_sq_cache`.
Refer to `stx_build_sq_cache`.

Reply:
I've re-designed that function. Please confirm.